File: hammer-border/src/main/java/dev/vini2003/hammer/border/impl/common/border/CubicWorldBorder.java
package dev.vini2003.hammer.border.impl.common.border;

public interface CubicWorldBorder {
	double getCenterY();
	
	double getBoundUp();
	
	double getBoundDown();
	
	boolean contains(double x, double y, double z);
	
	boolean contains(double x, double y, double z, double margin);
	
	void setCenter(double x, double y, double z);
	
	double getDistanceInsideBorder(double x, double y, double z);
}

File: hammer-border/src/main/kotlin/dev/vini2003/hammer/border/common/border/CubicWorldBorder.kt
package dev.vini2003.hammer.border.common.border

interface CubicWorldBorder {
	val centerY: Double
	
	val boundUp: Double
	
	val boundDown: Double
	
	fun contains(x: Double, y: Double, z: Double): Boolean
	
	fun contains(x: Double, y: Double, z: Double, margin: Double): Boolean
	
	fun setCenter(x: Double, y: Double, z: Double)
	
	fun getDistanceInsideBorder(x: Double, y: Double, z: Double): Double
}